pkg/cmd/server/server_test.go

```diff
@@ -578,3 +585,81 @@ func TestSupportOldAndNewReadReplicaConnectionPoolFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildMemoryProtectionConfig(t *testing.T) {
+	testcases := map[string]struct {
+		config       *Config
+		expectedErr  string
+		expectedType string
+	}{
+		`disabled`: {
+			config: &Config{
+				MemoryProtectionEnabled: false,
+			},
+			expectedType: "*memoryprotection.HarcodedMemoryLimitProvider",
+		},
+		`enabled`: {
+			config: &Config{
+				MemoryProtectionEnabled: true,
+			},
+			expectedType: "*memoryprotection.GoRealTimeMemoryLimiter",
+		},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			memoryUsageProvider := tc.config.buildMemoryUsageProvider()
+			require.NotNil(t, memoryUsageProvider)
+			require.Equal(t, tc.expectedType, reflect.TypeOf(memoryUsageProvider).String())
+		})
+	}
+}
+
+func TestBuildDispatchServer(t *testing.T) {
+	testcases := map[string]struct {
+		config                              *Config
+		expectedDispatchUnnaryMiddleware    int
+		expectedDispatchStreamingMiddleware int
+	}{
+		`auth:preshared key`: {
+			config: &Config{
+				PresharedSecureKey: []string{"securekey"},
+			},
+			expectedDispatchUnnaryMiddleware:    8,
+			expectedDispatchStreamingMiddleware: 6,
+		},
+		`auth:custom`: {
+			config: &Config{
+				GRPCAuthFunc: func(ctx context.Context) (context.Context, error) {
+					return ctx, nil
+				},
+			},
+			expectedDispatchUnnaryMiddleware:    8,
+			expectedDispatchStreamingMiddleware: 6,
+		},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockDatastore := mocks.NewMockDatastore(ctrl)
+			mockDispatcher := mocks.NewMockDispatcher(ctrl)
```

Comment:
I know mocks are in general contentious, but I don't think you're getting value out of having generated mocks for these vs.  just passing dummy objects of the right type or using the real objects. They're not really being used like mocks.